### PR TITLE
#2 Add basic org filtering functionality

### LINKED ISSUE #217: Change from challenge.archived to challenge.active
When the change was made: https://github.com/emfoundation/ce100-app/pull/199/files#diff-b00b2426ea4c5569a2f693c13130c545R10

I think the data booleans were not replaced with their opposite.  May need updating on your local branch @mk4111. Also wiki needs updating.

### PR DESCRIPTION
**Pull request by JMurphyWeb**
Add filter orgs by tag functionality  #215, remove extraneous filtering of challenge cards 
#19 #20 #192 #216 
resolves #217 

Some things to do that I will go onto address in the next PR:
covered loosely in #220 which i will pad out after i have deployed.

### CODE REVIEW

**Comment by JMurphyWeb** [comment]
@nelsonic sorry its all in one commit but the change caused some bugs so had to fix before committing.

**Comment by JMurphyWeb** [comment]
@mk4111 can you change base to master and assign @nelsonic if happy?

**Comment by mk4111** [comment]
:eyes: <3

File: server/handlers/all-users-view.js [hunk 1]
```diff
@@ -30,25 +30,6 @@ function filterActiveAndAdmin (arr) {
   return arr.filter(el => el.active && el.user_type !== 'admin');
```
Comment (mk4111) [inline server/handlers/all-users-view.js]: @JMurphyWeb - do you reckon we should put this in helpers? I don't think it needs to be done here (and might be premature optimisation if we do) but I can see where filterActive, filterAdmin, filterPrimary, etc. could be/is used in various places in our code source

File: server/handlers/helpers.js [hunk 2]
```diff
@@ -110,6 +110,24 @@ helpers.getTagNames = (tagIds) => {
     : false;
 };
 
+helpers.cloneArray = (arr) => arr.map(el => Object.assign({}, el));
+
+helpers.sortAlphabetically = (key) => (arr) =>
```
Comment (mk4111) [inline server/handlers/helpers.js]: @JMurphyWeb - is there a benefit of currying here instead of just passing in two arguments?

Comment (JMurphyWeb) [inline server/handlers/helpers.js]: We use this function in 2 ways, sort name and sort first_name.

I was thinking if we could partially apply it with for example first_name then export that as a function to be used in multiple places.

At the moment it doesn't really serve a purpose to curry. do you want me to change it to 2 args?

File: server/handlers/all-users-view.js (hunk 1 above)
Comment (JMurphyWeb) [inline server/handlers/all-users-view.js]: Not in this PR so will revisit when refactoring. @mk4111 will add to future enhancements in wiki.

File: server/handlers/helpers.js (hunk 2 above)
Comment (JMurphyWeb) [inline server/handlers/helpers.js]: Discussed with @mk4111 and agreed to park for now, will refactor in the future if there are not more occasions when we want to sort.

**Comment by JMurphyWeb** [comment]
@nelsonic Maries laptop is down so we have just reviewed her comments. On to you methinks.

**Comment by nelsonic** [comment]
Looks good. still have `/browse` in urls. we can remove in later PR. 👍

**Comment by JMurphyWeb** [comment]
OK great will issue

**Pull request merged**